src/Physics/Explodinator.cpp

```diff
@@ -214,59 +232,48 @@ namespace Explodinator
 			a_Chunk = Neighbour;
 
 			// Increment the simulation, weaken the ray:
-			Checkpoint += Step;
+			Checkpoint += a_Step;
 			a_Intensity -= StepAttenuation;
 		}
 	}
 
 	/** Sends out Explosion Lazors (tm) originating from the given position that destroy blocks. */
 	static void DamageBlocks(cChunk & a_Chunk, const Vector3f a_Position, const unsigned a_Power, const bool a_Fiery, const cEntity * const a_ExplodingEntity)
 	{
-		const auto Intensity = a_Power * (0.7f + GetRandomProvider().RandReal(0.6f));
-		const auto ExplosionRadius = CeilC((Intensity / StepAttenuation) * StepUnit);
 
 		// Oh boy... Better hope you have a hot cache, 'cos this little manoeuvre's gonna cost us 1352 raytraces in one tick...
-		const int HalfSide = TraceCubeSideLength / 2;
 
 		// The following loops implement the tracing algorithm described in http://minecraft.gamepedia.com/Explosion
 
 		// Trace rays from the explosion centre to all points in a square of area TraceCubeSideLength * TraceCubeSideLength
-		// in the YM and YP directions:
-		for (int OffsetX = -HalfSide; OffsetX < HalfSide; OffsetX++)
+		// Careful to avoid duplicates along edges
+		// Top and bottom sides:
+		for (float OffsetX = -1; OffsetX < 1; OffsetX+=GridPointSpacing)
```

Comment:
As far as I can tell vanilla behaviour is using `int`s here so we should keep it as before